### DISCUSSION TITLE
Call to undefined method Lcobucci\\JWT\\Token\\Plain::getClaim()

### DISCUSSION THREAD

**Discussion by AliHassan13** [kreait/firebase-php#531] [Q&A] [answered]
There is issue when i run the command `composer update` it's update the version of  **lcobucci/jwt** **4.0** it cause the issue and show the error
**Call to undefined method Lcobucci\\JWT\\Token\\Plain::getClaim()**
for the temporary fix solution you need to rollback the version of **lcobucci/jwt** to **3.4.1** by running the following command. 
**composer require lcobucci/jwt 3.4.1**

**firebase-php version 5.14.0**

**Comment by jeromegamez** [comment, marked as answer]
The issue is that `getClaim()` has been deprecated in 3.4 and removed in 4.0. I assume you're using the method somewhere in your code, but since you didn't share code, I can only guess :)

If you found the `getClaim()` somewhere in the code or docs of the SDK, please let me know or create a PR 🙏

**Reply by AliHassan13** [reply]
when i verify the token & get the uid of that user using this code.. 
```
try {
    $verifiedIdToken = $auth->verifyIdToken($idTokenString);
} catch (InvalidToken $e) {
    echo 'The token is invalid: '.$e->getMessage();
} catch (\InvalidArgumentException $e) {
    echo 'The token could not be parsed: '.$e->getMessage();
}

$uid = $verifiedIdToken->getClaim('sub');
$user = $auth->getUser($uid);
```

here i use **getClaim()** for the reference you can check this.

https://firebase-php.readthedocs.io/en/stable/authentication.html

**Reply by jeromegamez** [reply]
Thanks for the clarification! I forgot to update the docs for 5.14, but the will be correct with the next release (see https://github.com/kreait/firebase-php/commit/62dd2f1afb36b74ad750952f688865ae4a91fe88)

Replace

```php
$uid = $verifiedIdToken->getClaim('sub');
```
with 
```php
$uid = $verifiedIdToken->claims()->get('sub');
```
and you should be good to go 🤞

**Reply by denusklo** [reply]
I used 

`
$uid = $verifiedIdToken->claims()->get('sub');
`

but `claims()` method is undefined in my code.

**Comment by AliHassan13** [comment]
Thanks a lot..
